source/common/buffer/buffer_impl.cc

```diff
@@ -588,5 +588,35 @@ std::vector<Slice::SliceRepresentation> OwnedImpl::describeSlicesForTest() const
   return slices;
 }
 
+size_t OwnedImpl::addFragments(absl::Span<const absl::string_view> fragments) {
+  size_t total_size_to_write = 0;
+
+  for (const auto& fragment : fragments) {
+    total_size_to_write += fragment.size();
+  }
+
+  if (slices_.empty()) {
+    slices_.emplace_back(Slice(total_size_to_write, account_));
+  }
+
+  Slice& back = slices_.back();
+  if (back.reservableSize() >= total_size_to_write) {
+    uint8_t* mem = back.base_ + back.reservable_;
+    for (const auto& fragment : fragments) {
+      memcpy(mem, fragment.data(), fragment.size()); // NOLINT(safe-memcpy)
+      mem += fragment.size();
+    }
+    length_ += total_size_to_write;
+    back.reservable_ += total_size_to_write;
```

Comment:
Can we just use the existing API on Slice to create a reservation via `reserve()` memcpy the fragments into that and then do `back.commit(reservation)`?

Reply:
then there must be some more necessary check 🤔 It is minor update and I can try update it and check the benchmark result.

Reply:
look like it is ok.